checks/migrations/0019_hof.py

```diff
@@ -0,0 +1,127 @@
+            name="Fame",
+            fields=[
+                ("id", models.IntegerField()),
+                ("domain", models.CharField(max_length=255, primary_key=True)),
+                (
+                    "site_report",
+                    models.ForeignKey(
+                        null=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to="checks.domaintestreport",
+                        db_index=False,
+                    ),
+                ),
+                ("site_report_timestamp", models.DateTimeField(null=True)),
+                (
+                    "mail_report",
+                    models.ForeignKey(
+                        null=True,
+                        on_delete=django.db.models.deletion.CASCADE,
+                        to="checks.mailtestreport",
+                        db_index=False,
+                    ),
+                ),
+                ("mail_report_timestamp", models.DateTimeField(null=True)),
+            ],
+            options={
+                "abstract": False,
+            },
+        ),
+        # AutoField has to be primary key in Django, to solve this manually alter SQL this field
+        # see https://github.com/django/django/blob/787f3130f751283140fe2be8188eb5299552232d/django/db/models/fields/__init__.py#L2801
+        migrations.RunSQL(
+            sql=[
+                'ALTER TABLE "checks_fame" ALTER COLUMN "id" ADD GENERATED BY DEFAULT AS IDENTITY;',
+                """
+CREATE OR REPLACE FUNCTION fn_update_site_fame()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF NEW.score IS NULL THEN
+    -- NOOP
+  ELSIF NEW.score = 100 THEN
+    INSERT INTO checks_fame (domain, site_report_id, site_report_timestamp, mail_report_id, mail_report_timestamp)
+    VALUES (NEW.domain, NEW.id, NEW.timestamp, NULL, NULL)
+    ON CONFLICT (domain)
+    DO UPDATE SET site_report_id = NEW.id, site_report_timestamp = NEW.timestamp;
+  ELSE
+    DELETE FROM checks_fame WHERE domain = NEW.domain AND mail_report_id IS NULL;
+    UPDATE checks_fame SET site_report_id = NULL, site_report_timestamp = NULL WHERE domain = NEW.domain;
+  END IF;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+""",
+                """
+CREATE OR REPLACE FUNCTION fn_update_mail_fame()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF NEW.score IS NULL THEN
+    -- NOOP
+  ELSIF NEW.score = 100 THEN
+    INSERT INTO checks_fame (domain, site_report_id, site_report_timestamp, mail_report_id, mail_report_timestamp)
+      VALUES (NEW.domain, NULL, NULL, NEW.id, NEW.timestamp)
+    ON CONFLICT (domain)
+    DO UPDATE SET mail_report_id = NEW.id, mail_report_timestamp = NEW.timestamp;
+  ELSE
+    DELETE FROM checks_fame WHERE domain = NEW.domain AND site_report_id IS NULL;
+    UPDATE checks_fame SET mail_report_id = NULL, mail_report_timestamp = NULL WHERE domain = NEW.domain;
+  END IF;
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+""",
+                """
+CREATE TRIGGER tr_update_site_fame
```

Comment:
Can we do this in Python/Django ORM instead of Postgres? It's kind of hidden now in the migration, making it hard to discover and debug. Maybe using Django Signals https://docs.djangoproject.com/en/5.2/ref/signals/#post-save or something https://django-pgtrigger.readthedocs.io/en/4.15.2/ to represent it in Python code?

Reply:
I did consider django-pgtrigger, but that would introduce a package, and I would just try to re-create the desired trigger outcome by tinkering the Python. But of course let's ask @mxsasha what is best here.
Would some comment / documentation be sufficient?